File: app/local_plugins/ai_plugin_interface/android/src/main/kotlin/com/aiagent/plugin_interface/NativeAgentConfig.kt
package com.aiagent.plugin_interface

/**
 * Agent 初始化配置（从 Flutter MethodChannel 传入）
 */
data class NativeAgentConfig(
    val agentId: String,
    val inputMode: String,              // "text" | "short_voice" | "call"
    val sttVendor: String?,             // "azure"
    val ttsVendor: String?,             // "azure"
    val llmVendor: String?,             // "openai"
    val stsVendor: String?,             // "doubao"
    val astVendor: String?,             // "volcengine"
    val translationVendor: String?,     // "deepl" | "aliyun"
    val sttConfigJson: String?,
    val ttsConfigJson: String?,
    val llmConfigJson: String?,
    val stsConfigJson: String?,
    val astConfigJson: String?,
    val translationConfigJson: String?,
    val extraParams: Map<String, String> = emptyMap(),  // srcLang, dstLang etc.
) {
    companion object {
        /**
         * 从 Flutter MethodChannel 的 Map 构建
         */
        fun fromMap(map: Map<*, *>) = NativeAgentConfig(
            agentId = map["agentId"] as String,
            inputMode = map["inputMode"] as? String ?: "text",
            sttVendor = map["sttVendor"] as? String,
            ttsVendor = map["ttsVendor"] as? String,
            llmVendor = map["llmVendor"] as? String,
            stsVendor = map["stsVendor"] as? String,
            astVendor = map["astVendor"] as? String,
            translationVendor = map["translationVendor"] as? String,
            sttConfigJson = map["sttConfigJson"] as? String,
            ttsConfigJson = map["ttsConfigJson"] as? String,
            llmConfigJson = map["llmConfigJson"] as? String,
            stsConfigJson = map["stsConfigJson"] as? String,
            astConfigJson = map["astConfigJson"] as? String,
            translationConfigJson = map["translationConfigJson"] as? String,
            extraParams = (map["extraParams"] as? Map<*, *>)
                ?.map { (k, v) -> k.toString() to v.toString() }
                ?.toMap() ?: emptyMap(),
        )
    }
}

File: apps/client/local_plugins/core/ai_plugin_interface/ios/Classes/NativeAgentConfig.swift
import Foundation

/// Agent bootstrap config delivered over the Flutter MethodChannel.
public struct NativeAgentConfig {
    public let agentId: String
    /// "text" | "short_voice" | "call"
    public let inputMode: String

    public let sttVendor: String?
    public let ttsVendor: String?
    public let llmVendor: String?
    public let stsVendor: String?
    public let astVendor: String?
    public let translationVendor: String?

    public let sttConfigJson: String?
    public let ttsConfigJson: String?
    public let llmConfigJson: String?
    public let stsConfigJson: String?
    public let astConfigJson: String?
    public let translationConfigJson: String?

    /// MCP server list as a JSON array string (each entry = McpServerConfig.toJson).
    public let mcpServersJson: String?

    /// Free-form key/value extras (e.g. `srcLang`, `dstLang`).
    public let extraParams: [String: String]

    public init(
        agentId: String,
        inputMode: String = "text",
        sttVendor: String? = nil,
        ttsVendor: String? = nil,
        llmVendor: String? = nil,
        stsVendor: String? = nil,
        astVendor: String? = nil,
        translationVendor: String? = nil,
        sttConfigJson: String? = nil,
        ttsConfigJson: String? = nil,
        llmConfigJson: String? = nil,
        stsConfigJson: String? = nil,
        astConfigJson: String? = nil,
        translationConfigJson: String? = nil,
        mcpServersJson: String? = nil,
        extraParams: [String: String] = [:]
    ) {
        self.agentId = agentId
        self.inputMode = inputMode
        self.sttVendor = sttVendor
        self.ttsVendor = ttsVendor
        self.llmVendor = llmVendor
        self.stsVendor = stsVendor
        self.astVendor = astVendor
        self.translationVendor = translationVendor
        self.sttConfigJson = sttConfigJson
        self.ttsConfigJson = ttsConfigJson
        self.llmConfigJson = llmConfigJson
        self.stsConfigJson = stsConfigJson
        self.astConfigJson = astConfigJson
        self.translationConfigJson = translationConfigJson
        self.mcpServersJson = mcpServersJson
        self.extraParams = extraParams
    }

    /// Parse a MethodChannel argument dictionary into a config struct.
    /// `agentId` is required; missing/empty values fall through to `nil`.
    public static func fromMap(_ map: [String: Any?]) -> NativeAgentConfig {
        // Force-unwrap mirrors the Kotlin `as String` cast — the Dart side
        // guarantees agentId is set when invoking createAgent.
        let agentId = (map["agentId"] as? String) ?? ""
        let extras: [String: String]
        if let raw = map["extraParams"] as? [String: Any?] {
            extras = raw.reduce(into: [String: String]()) { acc, kv in
                if let value = kv.value {
                    acc[kv.key] = String(describing: value)
                }
            }
        } else {
            extras = [:]
        }
        return NativeAgentConfig(
            agentId: agentId,
            inputMode: (map["inputMode"] as? String) ?? "text",
            sttVendor: map["sttVendor"] as? String,
            ttsVendor: map["ttsVendor"] as? String,
            llmVendor: map["llmVendor"] as? String,
            stsVendor: map["stsVendor"] as? String,
            astVendor: map["astVendor"] as? String,
            translationVendor: map["translationVendor"] as? String,
            sttConfigJson: map["sttConfigJson"] as? String,
            ttsConfigJson: map["ttsConfigJson"] as? String,
            llmConfigJson: map["llmConfigJson"] as? String,
            stsConfigJson: map["stsConfigJson"] as? String,
            astConfigJson: map["astConfigJson"] as? String,
            translationConfigJson: map["translationConfigJson"] as? String,
            mcpServersJson: map["mcpServersJson"] as? String,
            extraParams: extras
        )
    }
}
